pkg/gardenlet/controller/networkpolicy/add.go

```diff
@@ -154,6 +154,10 @@ func (r *Reconciler) ClusterPredicate() predicate.Predicate {
 				return false
 			}
 
+			if shoot.Spec.Networking == nil || oldShoot.Spec.Networking == nil {
```

Comment:
For a workerless shoot `Services` and `IPFamilies` can still be set. So imaging that in the old shoot nothing was set. But then an update was made to `Services` and/or `IPFamilies` in which case `shoot.Spec.Networking` will not be nil. However since this is an OR condition this will return false. Is this desired?

Reply:
Add test @ary1992